includes/discovery/processors/edgecos.inc.php

```diff
@@ -13,10 +13,38 @@
 if ($device['os'] == 'edgecos') {
     echo 'Edgecos : ';
 
-    $oid = '.1.3.6.1.4.1.259.10.1.24.1.39.2.1.0';
-    $descr = 'Processor';
-    $usage = snmp_get($device, $oid, '-Ovqn');
+    $usage = snmp_get($device, '.1.3.6.1.4.1.259.10.1.24.1.39.2.1.0', '-Ovqn');//ECS4510
```

Comment:
Same goes for this file really, no point in running snmp_gets all the time, just check if `$oid` is set before you do.

Check you can't sysObjectID to work these out as well. Will save a lot of code and snmp queries.